ath79: add syscon compatible property to ar7100 ethernet nodes

tofurky opened the pull request:
This adds "syscon" to the compatible properties for the eth0/eth1 nodes
in ar7100.dtsi.

Without this, a kernel panic is encountered on boot with some ar7100
boards. This for some reason wasn't an issue for the WNDR3800, which
uses a Realtek switch chipset, but the panic was encountered on the
RouterStation Pro (using an AR8216 switch) and some other boards that
haven't yet been merged.

The panic message mentions an unaligned access and happens in
ag71xx_mdio_probe in drivers/net/ethernet/atheros/ag71xx/ag71xx_mdio.c.

Even if the unaligned access is fixed, the ag71xx_mdio probe still fails
without the "syscon" property.

This was already being worked around in
ar7161_ubnt_routerstation-pro.dts by overriding the compatible property,
so this commit removes that as well.

All of the other ath79 .dtsi already have this property, so no changes
are needed elsewhere.

Signed-off-by: Matt Merhar <mattmerhar@protonmail.com>

tofurky committed:
ath79: add syscon compatible property to ar7100 ethernet nodes

This adds "syscon" to the compatible properties for the eth0/eth1 nodes
in ar7100.dtsi.

Without this, a kernel panic is encountered on boot with some ar7100
boards. This for some reason wasn't an issue for the WNDR3800, which
uses a Realtek switch chipset, but the panic was encountered on the
RouterStation Pro (using an AR8216 switch) and some other boards that
haven't yet been merged.

The panic message mentions an unaligned access and happens in
ag71xx_mdio_probe in drivers/net/ethernet/atheros/ag71xx/ag71xx_mdio.c.

Even if the unaligned access is fixed, the ag71xx_mdio probe still fails
without the "syscon" property.

This was already being worked around in
ar7161_ubnt_routerstation-pro.dts by overriding the compatible property,
so this commit removes that as well.

All of the other ath79 .dtsi already have this property, so no changes
are needed elsewhere.

Signed-off-by: Matt Merhar <mattmerhar@protonmail.com>

hanipouspilot commented on target/linux/ath79/dts/ar7161_ubnt_routerstation-pro.dts:
```diff
@@ -21,13 +21,11 @@
 };
 
 &eth1 {
-	compatible = "qca,ar7100-eth", "syscon";
```
We don't need to add it again here if you add it to ar7100.dtsi

tofurky commented:
yes, it was posted in this thread but i didn't author it so i didn't include it. i mentioned this PR and asked if they could submit it in another PR: https://forum.openwrt.org/t/is-anybody-working-on-linux-4-14-for-ar71xx-platform-porting-guide-to-ath79/13013/765

@drvlabo do you want me to submit it? i can add you to the signed-off-by if so.

drvlabo commented:
@tofurky please PR it instead of me.

tofurky commented:
@drvlabo okay sure, but can i attribute the patch to you somehow? do you have a signed-off-by: first last <email@somewhere.com> i can add?

drvlabo commented:
Add your own signed-off-by, please. I don't mind.

2018年8月30日(木) 19:35 tofurky <notifications@github.com>:

> @drvlabo <https://github.com/drvlabo> okay sure, but can i attribute the
> patch to you somehow? do you have a signed-off-by: first last
> email@somewhere.com i can add?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/1340#issuecomment-417273066>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AivPTygatDrrobY8scLd_h6TfZDxVf4Kks5uV7_wgaJpZM4WSl2D>
> .
>

tofurky commented:
okay thanks @drvlabo, i added a second commit with the patch.
i wasn't sure about the naming but of course it's a platform patch. the ones closest in series are:
```
...
490-usb-ehci-add-quirks-for-qca-socs.patch
900-mdio_bitbang_ignore_ta_value.patch
901-phy-mdio-bitbang-prevent-rescheduling-during-command.patch
902-ag71xx-mdio-probe-unaligned-panic.patch
910-unaligned_access_hacks.patch

```

981213 commented:
You should patch ag71xx driver directly in target/linux/ath79/files instead of creating a separated patch to do this :)

tofurky commented:
oops yeah i'll fix that, thanks.

tofurky committed:
ath79: fix unaligned access panic in ag71xx_mdio_probe

Without "syscon" being present in an ag71xx ethernet DT node's
compatible property, a panic occurs at boot during probe citing
"Unhandled kernel unaligned access".

With this modification, the panic no longer occurs and instead the probe
simply fails, allowing the boot process to continue.

Signed-off-by: Matt Merhar <mattmerhar@protonmail.com>

tofurky commented:
okay, fixed. i'd incorrectly assumed that the driver was part of mainline, as i didn't see it in target/linux/\*/patches\*.

mkresin commented:
merged, thanks!
